independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/DependentContext.java

```diff
@@ -0,0 +1,60 @@
+package io.quarkus.arc.impl;
+
+import java.lang.annotation.Annotation;
+
+import jakarta.enterprise.context.Dependent;
+import jakarta.enterprise.context.spi.Contextual;
+import jakarta.enterprise.context.spi.CreationalContext;
+
+import io.quarkus.arc.InjectableBean;
+import io.quarkus.arc.InjectableContext;
+
+/**
+ * The built-in context for {@link jakarta.enterprise.context.Dependent}.
+ */
+class DependentContext implements InjectableContext {
+    @Override
+    public Class<? extends Annotation> getScope() {
+        return Dependent.class;
+    }
+
+    @Override
+    public <T> T get(Contextual<T> contextual, CreationalContext<T> creationalContext) {
+        if (creationalContext == null) {
+            return null;
```

Comment:
I am probably missing something obvious but why is it ok to have the cc param `null` and return `null` as well?

Reply:
Also see https://github.com/weld/core/blob/master/impl/src/main/java/org/jboss/weld/contexts/unbound/DependentContextImpl.java :-)

Reply:
Ah, right, thanks!